### PR TITLE
Woodpecker rust 1.75

### LINKED ISSUE #4354: [Bug]: Docker build error "future cannot be sent between threads safely"
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

Docker builds have recently started to throw the following error:
```
#28 336.2 error: future cannot be sent between threads safely
#28 336.2    --> crates/federate/src/util.rs:64:45
#28 336.2     |
#28 336.2 64  |       let task: JoinHandle<()> = tokio::spawn(async move {
#28 336.2     |  _____________________________________________^
#28 336.2 65  | |       loop {
#28 336.2 66  | |         let res = task(stop2.clone()).await;
#28 336.2 67  | |         if stop2.is_cancelled() {
#28 336.2 ...   |
#28 336.2 72  | |       }
#28 336.2 73  | |     });
#28 336.2     | |_____^ future created by async block is not `Send`
#28 336.2     |
#28 336.2 note: future is not `Send` as this value is used across an await
#28 336.2    --> crates/federate/src/util.rs:66:39
#28 336.2     |
#28 336.2 66  |         let res = task(stop2.clone()).await;
#28 336.2     |                   ----                ^^^^^- `task` is later dropped here
#28 336.2     |                   |                   |
#28 336.2     |                   |                   await occurs here, with `task` maybe used later
#28 336.2     |                   has type `&impl Fn(CancellationToken) -> F + Send + 'static` which is not `Send`
#28 336.2 help: consider moving this into a `let` binding to create a shorter lived borrow
#28 336.2    --> crates/federate/src/util.rs:66:19
#28 336.2     |
#28 336.2 66  |         let res = task(stop2.clone()).await;
#28 336.2     |                   ^^^^^^^^^^^^^^^^^^^
#28 336.2 note: required by a bound in `tokio::spawn`
#28 336.2    --> /home/lemmy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.0/src/task/spawn.rs:166:21
#28 336.2     |
#28 336.2 164 |     pub fn spawn<F>(future: F) -> JoinHandle<F::Output>
#28 336.2     |            ----- required by a bound in this function
#28 336.2 165 |     where
#28 336.2 166 |         F: Future + Send + 'static,
#28 336.2     |                     ^^^^ required by this bound in `spawn`
#28 336.2 help: consider further restricting this bound
#28 336.2     |
#28 336.2 57  |     task: impl Fn(CancellationToken) -> F + Send + 'static + std::marker::Sync,
#28 336.2     |                                                            +++++++++++++++++++
#28 336.2 
#28 336.3 error: could not compile `lemmy_federate` (lib) due to previous error
#28 336.3 warning: build failed, waiting for other jobs to finish...
#28 ERROR: process "/bin/sh -c set -ex;     if [ \"${RUST_RELEASE_MODE}\" = \"release\" ]; then         echo \"pub const VERSION: &str = \\\"$(git describe --tag)\\\";\" > crates/utils/src/version.rs;         [ -z \"$USE_RELEASE_CACHE\" ] && cargo clean --release;         cargo build --features \"${CARGO_BUILD_FEATURES}\" --release;         mv \"./target/$CARGO_BUILD_TARGET/$RUST_RELEASE_MODE/lemmy_server\" /home/lemmy/lemmy_server;     fi" did not complete successfully: exit code: 101
------
 > [linux/amd64 build-arm64 5/5] RUN --mount=type=cache,target=./target,uid=10001,gid=10001 set -ex;     if [ "release" = "release" ]; then         echo "pub const VERSION: &str = "$(git describe --tag)";" > crates/utils/src/version.rs;         [ -z "$USE_RELEASE_CACHE" ] && cargo clean --release;         cargo build --features "default" --release;         mv "./target/aarch64-unknown-linux-gnu/release/lemmy_server" /home/lemmy/lemmy_server;     fi:
336.2 165 |     where
336.2 166 |         F: Future + Send + 'static,
336.2     |                     ^^^^ required by this bound in `spawn`
336.2 help: consider further restricting this bound
336.2     |
336.2 57  |     task: impl Fn(CancellationToken) -> F + Send + 'static + std::marker::Sync,
336.2     |                                                            +++++++++++++++++++
336.2 
336.3 error: could not compile `lemmy_federate` (lib) due to previous error
336.3 warning: build failed, waiting for other jobs to finish...
------
Dockerfile:69
--------------------
  68 |     # Release build
  69 | >>> RUN --mount=type=cache,target=./target,uid=10001,gid=10001 set -ex; \
  70 | >>>     if [ "${RUST_RELEASE_MODE}" = "release" ]; then \
  71 | >>>         echo "pub const VERSION: &str = \"$(git describe --tag)\";" > crates/utils/src/version.rs; \
  72 | >>>         [ -z "$USE_RELEASE_CACHE" ] && cargo clean --release; \
  73 | >>>         cargo build --features "${CARGO_BUILD_FEATURES}" --release; \
  74 | >>>         mv "./target/$CARGO_BUILD_TARGET/$RUST_RELEASE_MODE/lemmy_server" /home/lemmy/lemmy_server; \
  75 | >>>     fi
  76 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c set -ex;     if [ \"${RUST_RELEASE_MODE}\" = \"release\" ]; then         echo \"pub const VERSION: &str = \\\"$(git describe --tag)\\\";\" > crates/utils/src/version.rs;         [ -z \"$USE_RELEASE_CACHE\" ] && cargo clean --release;         cargo build --features \"${CARGO_BUILD_FEATURES}\" --release;         mv \"./target/$CARGO_BUILD_TARGET/$RUST_RELEASE_MODE/lemmy_server\" /home/lemmy/lemmy_server;     fi" did not complete successfully: exit code: 101
time="2024-01-05T12:31:02Z" level=error msg="execution failed: exit status 1"
```

[Link to CI logs](https://woodpecker.join-lemmy.org/repos/129/pipeline/4934/22#L2852)

Most importantly this prevents us from making new releases. The problem is quite inconsistent, it only seems to happen in Docker builds but not native builds. In local Docker builds I could reproduce it with Rust 1.72.1, but it went away after upgrading to 1.75 (https://github.com/LemmyNet/lemmy/pull/4353). However in CI its still throwing the same error.

cc @phiresky

### PR DESCRIPTION
**Pull request by dessalines**
This comments out the docker arm builder until this PR is ready, and rasky gets another arm image up: https://github.com/raskyld/lemmy-cross-toolchains/pull/1

Edit: rasky got the updated arm image out, so this fixes it now. 

Fixes #4354

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -41,38 +41,38 @@ RUN --mount=type=cache,target=/lemmy/target set -ex; \
     fi
 
 # ARM64 builder
```
Comment (dessalines) [inline docker/Dockerfile:43]: Commenting out the broken arm builds until we get an updated rust arm image from @raskyld

Comment (raskyld) [reply docker/Dockerfile:43]: Hey! Thanks for pinging me, I launched a build with v0.2.0 tag :)

Comment (dessalines) [reply docker/Dockerfile:43]: Thx! I'll get another tag started to test it shortly.

**Comment by dessalines** [comment]
Okay this one passed! https://woodpecker.join-lemmy.org/repos/129/pipeline/4966/22 

Thanks @raskyld !

File: docker/Dockerfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ ARG CARGO_BUILD_FEATURES=default
 ARG RUST_RELEASE_MODE=debug
 
 ARG AMD_BUILDER_IMAGE=rust:${RUST_VERSION}
-ARG ARM_BUILDER_IMAGE="ghcr.io/raskyld/aarch64-lemmy-linux-gnu:v0.1.0"
+ARG ARM_BUILDER_IMAGE="ghcr.io/raskyld/aarch64-lemmy-linux-gnu:v0.2.0"
```
Comment (Nutomic) [inline docker/Dockerfile:7]: Would be good if you add a link to the github repo here.

**Review by Nutomic — APPROVED** [review]

File: docker/Dockerfile (hunk 2 above)
Comment (dessalines) [reply docker/Dockerfile:7]: Done

**Pull request merged**